bin/migrate-oats-data/documents/post_launch/oats_documents_to_alcs_documents_planning_review.py

```diff
@@ -0,0 +1,189 @@
+                            "system",
+                            uploaded_at,
+                            source           
+                        )
+                        VALUES{documents_to_insert}
+                        ON CONFLICT (oats_document_id) DO UPDATE SET 
+                            oats_document_id = EXCLUDED.oats_document_id, 
+                            file_name = EXCLUDED.file_name, 
+                            oats_planning_review_id = EXCLUDED.oats_planning_review_id, 
+                            audit_created_by = EXCLUDED.audit_created_by, 
+                            file_key = EXCLUDED.file_key, 
+                            mime_type = EXCLUDED.mime_type,
+                            tags = EXCLUDED.tags,
+                            "system" = EXCLUDED."system",
+                            uploaded_at = EXCLUDED.uploaded_at,
+                            source = EXCLUDED.source;
+    """
+
+
+def _prepare_data_to_insert(rows):
+    row_without_last_element = []
+    for row in rows:
+        mapped_row = _map_data(row)
+        row_without_last_element.append(tuple(mapped_row.values()))
+
+    return row_without_last_element
+
+
+def _map_data(row):
+    return {
+        "oats_document_id": row["oats_document_id"],
+        "file_name": row["file_name"],
+        "oats_planning_review_id": row["oats_planning_review_id"],
+        "audit_created_by": OATS_ETL_USER,
+        "file_key": row["file_key"],
+        "mime_type": _get_mime_type(row),
+        "tags": row["tags"],
+        "system": _map_system(row),
+        "file_upload_date": _get_upload_date(row),
+        "file_source": _get_document_source(row),
+    }
+
+
+def _map_system(row):
+    who_created = row["who_created"]
+    if who_created in ("PROXY_OATS_LOCGOV", "PROXY_OATS_APPLICANT"):
+        sys = "OATS_P"
+    else:
+        sys = "OATS"
+    return sys
+
+
+def _get_upload_date(data):
+    upload_date = data.get("uploaded_date", "")
+    created_date = data.get("when_created", "")
+    if upload_date:
+        return add_timezone_and_keep_date_part(upload_date)
+    else:
+        return add_timezone_and_keep_date_part(created_date)
+
+
+def _get_document_source(data):
+    source = data.get("document_source_code", "")
+    if source:
+        source = str(OatsToAlcsDocumentSourceCode[source].value)
+
+    return source
+
+
+def _get_mime_type(data):
+    file_name = data.get("file_name", "")
+    extension = os.path.splitext(file_name)[-1].lower().strip()
+    if extension == ".pdf":
+        return "planning_review/pdf"
+    else:
+        return "planning_review/octet-stream"
+
+
+@inject_conn_pool
+def document_pr_clean(conn=None):
+    logger.info("Start planning review related documents cleaning")
+    with conn.cursor() as cursor:
+        cursor.execute(
+            f"DELETE FROM alcs.document WHERE audit_created_by = '{OATS_ETL_USER}' AND oats_planning_review_id IS NOT NULL AND audit_created_at > '2024-02-08';"
```

Comment:
I do not think you need the audit_created_at filter since there are no PR documents on prod and staff is not using feature till the ETL is complete

Reply:
left as a safety it was left in case files were manually added, can be removed